tests/transformers/test_transformers_model_export.py

```diff
@@ -98,7 +96,14 @@ def force_gc():
 
 @pytest.fixture
 def model_path(tmp_path):
-    return tmp_path.joinpath("model")
+    model_path = tmp_path.joinpath("model")
+    yield model_path
+
+    # Pytest keeps the temporary directory created by `tmp_path` fixture for 3 recent test sessions
+    # by default. This is useful for debugging during local testing, but in CI it just wastes the
+    # disk space.
+    if os.getenv("GITHUB_ACTIONS") == "true":
+        shutil.rmtree(model_path, ignore_errors=True)
```

Comment:
Can we use https://docs.pytest.org/en/7.3.x/reference/reference.html#confval-tmp_path_retention_count or https://docs.pytest.org/en/7.3.x/reference/reference.html#confval-tmp_path_retention_policy instead?

Reply:
Let's use `tmp_path_retention_policy= failed`.

Reply:
which autolog test?

Reply:
https://github.com/mlflow/mlflow/actions/runs/7912208000/job/21597613197

Very weird error, looks irrelevant but happens when we clean up tmp_path somehow

Reply:
I was able to reproduce the same error locally.

```
tests/transformers/test_transformers_autolog.py::test_transformers_autolog_adheres_to_global_behavior_using_setfit PASSED | MEM 2.6/15.6 GB | DISK 42.5/72.5 GB [ 30%]
tests/transformers/test_transformers_autolog.py::test_transformers_autolog_adheres_to_global_behavior_using_trainer FAILED | MEM 2.7/15.6 GB | DISK 42.5/72.5 GB [ 40%]
```

We have two tests that start with the same prefix `test_transformers_autolog_adhe`. pytest creates a directory with name `test_transformers_autolog_adhe0` (first 30 characters + number) for them.

When `test_transformers_autolog_adheres_to_global_behavior_using_setfit` succeeds, pytest deletes `test_transformers_autolog_adhe0/mlruns.sqlite`. The same sqlite database file is used in `test_transformers_autolog_adheres_to_global_behavior_using_setfit`, but this somehow gets sqlite confused and sqlite complains cannot write.

Reply:
How pytest determines the temporary directory name:

https://github.com/pytest-dev/pytest/blob/5d7a5a9343ec7d675a9599312f8055e2a81cc1db/src/_pytest/tmpdir.py#L251

Reply:
Found the issue I filed last year: https://github.com/sqlalchemy/sqlalchemy/issues/9186

Reply:
Solution: #11153

Reply:
Ah interesting, thanks for the diving deep! Yea look like the connection is not reset even when the DB file is wiped out, then complain the file can't be opened as if it is permission issue.

Reply:
@harupy It seems the similar hard-coded DB name is used in many places (https://github.com/search?q=repo%3Amlflow%2Fmlflow+sqlite+path%3A%2F%5Etests%5C%2F%2F&type=code&p=2) and more flavor tests may fail by adding tmp path clean up. The effort of fixing all of them seems not to be worth as we only have this issue in Transformer tests.

I think we have two options 
1. Set the `tmp_path_retention_policy` option in `tests/transformers/conftest.py`.
2. Just clean up in model path fixture.

Both should solve the issue, but I'm slighly inclined to the latter option, because overriding the setting in another `pytest_configure` in transformer's conftest could be confusing due to low visibility. WDYT?

Reply:
Makes sense, let's do the second option!